src/services/WaveformDataUtils.js

```diff
@@ -259,40 +303,30 @@ export default class WaveformDataUtils {
   /**
    * Prevent the times of segment being edited overlapping with the existing segments
    * @param {Object} segment - segement being edited in the waveform
+   * @param {Boolean} inMarker - true -> start time changed, false -> end time changed
    * @param {Object} peaksInstance - current peaks instance for waveform
    */
-  validateSegment(segment, peaksInstance) {
-    const allSegments = peaksInstance.segments.getSegments();
+  validateSegment(segment, inMarker, peaksInstance) {
     const duration = this.roundOff(peaksInstance.player.getDuration());
 
     const { startTime, endTime } = segment;
 
     // segments before and after the editing segment
-    const { before, after } = this.findWrapperSegments(segment, allSegments);
+    const { before, after } = this.findWrapperSegments(segment, peaksInstance);
 
-    for (let i = 0; i < allSegments.length; i++) {
-      const current = allSegments[i];
-      if (current.id == segment.id) {
-        continue;
+    if (inMarker) {
+      if (before && startTime < before.endTime) {
+        segment.update({ startTime: before.endTime });
```

Comment:
Am I reading this correctly that if you try to make the start time earlier then it will force it back to before.endTime?

Reply:
Yes, the handles won't go pass the endTime of the segment before.